src/components/Logs.tsx

```diff
@@ -1,25 +1,57 @@
 import { MutinyWallet } from "@mutinywallet/mutiny-wasm";
+import { createSignal, Show } from "solid-js";
 
-import { Button, InnerCard, NiceP, VStack } from "~/components";
+import {
+    Button,
+    InfoBox,
+    InnerCard,
+    NiceP,
+    SimpleDialog,
+    TextField,
+    VStack
+} from "~/components";
 import { useI18n } from "~/i18n/context";
-import { downloadTextFile } from "~/utils";
+import { downloadTextFile, eify } from "~/utils";
 
 export function Logs() {
     const i18n = useI18n();
+
+    // Create state for errors, password and dialog visibility
+    const [error, setError] = createSignal<Error>();
+    const [exportDecrypt, setExportDecrypt] = createSignal(false);
+    const [password, setPassword] = createSignal("");
+
     async function handleSave() {
         try {
-            const logs = await MutinyWallet.get_logs();
-
+            setError(undefined);
+            const logs = await MutinyWallet.get_logs(password()); // Use password if required
```

Comment:
I'm not totally sure what the diff between `get_logs("")` and `get_logs(undefined)` is (the other decrypt logic only runs when a password is needed, so it's always a string). but I'm pretty sure in case of a `None` you want `undefined`, not `""`

`const [password, setPassword] = createSignal();` will initialize `password()` to undefined. and you can clear it on line 47 `setPassword(undefined)`

Reply:
We handle both as the same because the empty string / undefined behavior is such a mess in javascript. So what you're suggesting is I just change one line? That's fine.